src/ondisk.rs

```diff
@@ -4396,6 +4733,122 @@ Clone)]
         }
     }
 
+    make_accessors! {
+        #[derive(FromBytes, AsBytes, Unaligned, PartialEq, Debug, Copy, Clone)]
+        #[repr(C, packed)]
+        pub struct DdrDqPinMapElementLane {
+            pins: [u8; 8],
+        }
+    }
+
+    impl DdrDqPinMapElementLane {
+        pub fn new(pins: [u8; 8]) -> Self {
+            Self { pins }
+        }
+    }
+
+    // XXX: Not that useful.
+    impl Default for DdrDqPinMapElementLane {
```

Comment:
No. It's a design limitation of the `impl_struct_serde_conversion` macro.

The reason is if we have a struct like this

   struct X {
     a: u32,
     b: u32,
   }

... and then build an instance up like this (which is what `impl_struct_serde_conversion` does):

    X::builder().with_a(3).with_b(4).build()

... it actually returns an instance of `X` at `X::builder()`. That means it has to default it--what would be the values of `a` and `b` otherwise?

This could be avoided by having some smarter intermediate builder instance, but I think it would be hard to do that without allocating--and most of the time you really DO want defaults anyway.

Just in this lane example you actually want all the pins in `DdrDqPinMapElement` to be unique (regardless of in which lane they are used). So in this one case the default is actually bad.